exporter/kinesisexporter/producer.go

```diff
@@ -0,0 +1,78 @@
+package kinesisexporter
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials/stscreds"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/kinesis"
+	omnition "github.com/signalfx/omnition-kinesis-producer"
+	"github.com/signalfx/omnition-kinesis-producer/loggers/kpzap"
+	"go.uber.org/zap"
+	"time"
+)
+
+type producer interface {
+	start()
+	stop()
+	put(data []byte, partitionKey string) error
+}
+
+type client struct {
+	client *omnition.Producer
+	logger *zap.Logger
+}
+
+func newKinesisProducer(c *Config, logger *zap.Logger) (producer, error) {
+	awsConfig := aws.NewConfig().WithRegion(c.AWS.Region).WithEndpoint(c.AWS.KinesisEndpoint)
+	sess, err := session.NewSession(awsConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	// If AWS role is provided, use sts credentials to assume the role
+	if len(c.AWS.Role) > 0 {
+		creds := stscreds.NewCredentials(sess, c.AWS.Role)
+		awsConfig = awsConfig.WithCredentials(creds)
```

Comment:
I think we need to pass this config into the kinesis client constructor on line 40. Otherwise this config isn't used.

Reply:
Good catch thanks!